### PR TITLE
V6 rename themes

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview
Rename the themes

## Screenshots (if applicable)
n/a

## Testing

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/theme/stories/index.stories.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ import Box from 'ui-box'
 import Component from '@reactions/component'
 import { Heading } from '../../typography'
 import { Checkbox } from '../../checkbox'
-import v5Theme from '../src/themes/v5'
-import v6Theme from '../src/themes/v6'
+import v5Theme from '../src/themes/legacy'
```
Comment (mshwery) [inline src/theme/stories/index.stories.js:7]: would suggest avoiding calling this "legacy" – might be better to call keep it as `v5`. wdyt?

Comment (colinlohner) [reply src/theme/stories/index.stories.js:7]: In my mind it was a subtle hint that we wont support it for very long. But can go with v5 if you think that's better

Comment (mshwery) [reply src/theme/stories/index.stories.js:7]: why wouldnt we support it? it's a valid theme according to the new theming capabilities, right?

Comment (colinlohner) [reply src/theme/stories/index.stories.js:7]: Just thinking we don't want to have to maintain two themes long term. If we added new components etc in the future we would have to update both themes

**Pull request merged**